WooCommerce/Classes/ViewRelated/Products/ProductSelector/ProductSelectorViewModel.swift

```diff
@@ -395,9 +395,9 @@ private extension ProductSelectorViewModel {
             return
         }
         for id in initialSelectedItems {
-            if products.contains(where: { $0.productID == id }) {
+            if products.contains(where: { $0.productID == id }), !selectedProductIDs.contains(id) {
                 selectedProductIDs.append(id)
-            } else {
+            } else if !selectedProductVariationIDs.contains(id) {
                 selectedProductVariationIDs.append(id)
             }
```

Comment:
Small issue with this change: If you select a product (not a variation) but `selectedProductIDs` already contains the ID, the product ID is added to `selectedProductVariationIDs` instead.

One option could be to separate the check to make sure the ID hasn't been added to one of the selected lists, maybe something like this:

```swift
            if selectedProductIDs.contains(id) || selectedProductVariationIDs.contains(id) {
                break
            } else if products.contains(where: { $0.productID == id }) {
                selectedProductIDs.append(id)
            } else {
                selectedProductVariationIDs.append(id)
            }
```

Reply:
Thank you for noticing the issue! I've made a change in 7f11b9b0f72b317bf2aee8eecd07cb9929f3d121 - I believe we should use `continue` instead of `break` because we want to traverse the whole list just in case any item is missing.